### PR TITLE
log outlier handling

### PR DESCRIPTION
**Pull request by fparisotto**

### CODE REVIEW

File: src/main/scala/ignition/core/jobs/utils/RDDUtils.scala [hunk 1]
```diff
@@ -11,6 +13,9 @@ import org.joda.time.format.DateTimeFormat
 import scalaz.{Success, Validation}
 
 object RDDUtils {
+
+  private lazy val logger = LoggerFactory.getLogger("ignition.RDDUtils")
```
Comment (douglaz) [inline src/main/scala/ignition/core/jobs/utils/RDDUtils.scala]: ignition.core.RDDUtils

File: src/main/scala/ignition/core/jobs/utils/RDDUtils.scala [hunk 2]
```diff
@@ -78,8 +83,24 @@ object RDDUtils {
     // TODO: add an way to log if we reach the limit
     def groupByKeyAndTake(n: Int): RDD[(K, List[V])] =
       rdd.aggregateByKey(List.empty[V])(
-        (lst, v) => if (lst.size >= n) lst else v :: lst,
-        (lstA, lstB) => if (lstA.size >= n) lstA else if (lstB.size >= n) lstB else (lstA ++ lstB).take(n)
+        (lst, v) =>
+          if (lst.size >= n) {
+            logger.warn(s"Ignoring value '$v' due aggregation result of size '${lst.size}' is bigger then n = '$n'")
```
Comment (douglaz) [inline src/main/scala/ignition/core/jobs/utils/RDDUtils.scala]: Use scala-logging or use the "{} {} {}", arg1 ,arg2, arg3 syntax

File: src/main/scala/ignition/core/jobs/utils/RDDUtils.scala [hunk 3]
```diff
@@ -78,8 +83,24 @@ object RDDUtils {
...
+          },
+        (lstA, lstB) =>
+          if (lstA.size >= n) {
+            logger.warn(s"First partition crossed the threshold, ignoring ${lstB.size} rows of second partition")
+            lstA
+          } else if (lstB.size >= n) {
+            logger.warn(s"Second partition crossed the threshold, ignoring ${lstA.size} rows of first partition")
+            lstB
+          } else {
+            logger.warn(s"Merging partition1={${lstA.size}} with partition2={${lstB.size}} and taking the first n={$n}")
```
Comment (douglaz) [inline src/main/scala/ignition/core/jobs/utils/RDDUtils.scala]: Can't be a warn here.

Comment (douglaz) [inline src/main/scala/ignition/core/jobs/utils/RDDUtils.scala]: Take a sample.

**Comment by fparisotto** [comment]
LGT @douglaz

**Pull request merged**